### PR TITLE
layers: Fix validation of vkCmdBindDescriptorSets() during renderpass

### PR DESCRIPTION
**Pull request by tobine**
Spec allows vkCmdBindDescriptorSets() to be called inside or outside of a
renderpass so killing two core_validation checks that violate this behavior.

NOTE : This GitHub diff on this is ridiculous. Just removed two checks and reduced indentation for a big block.

### CODE REVIEW

**Comment by lentinem** [comment]
+1 as far as a I can seems fine but it is hard to parse.

**Comment by tobine** [comment]
fix pushed to master
